src/ert/gui/simulation/view/realization.py

```diff
@@ -140,3 +148,16 @@ def _paint_inner_grid(self, painter: QPainter, rect: QRect, colors) -> None:
 
     def sizeHint(self, option, index) -> QSize:
         return self._size
+
+
+def determine_realization_status_color(job_colors) -> QColor:
+    states = [
+        COLOR_UNKNOWN,
+        COLOR_FINISHED,
+        COLOR_WAITING,
+        COLOR_PENDING,
+        COLOR_RUNNING,
+        COLOR_FAILED,
+    ]
+
+    return max(job_colors, key=lambda x: states.index(x), default=COLOR_UNKNOWN)
```

Comment:
Think pylint complains because one may replace lambda by `key = states.index`.

Reply:
Then Ill fix that too. 👍